- NAMUGENYI LISA LUSINGA
- S24B38/010
- B30300

# THEORY SECTION

- i. Shapiro-Wilk's test 
  - I would use this test to find out whether the population distribution by region is normal.  To find out if the population is normally distributed in all regions of Uganda.

- ii. One-way ANOVA test
    - I would use this test to compare the average household sizes in Uganda, especially those of different regions. This will help us know if there is a significant difference bewteen the household sizes of the regions of Uganda.

- iii. Pearson's correlation test
    - This would be used to find out how the population by age group affected the regional population growth. This way, the people in charge can plan for an area that has more youth to create employment opportunities, to prevent them from being idol and starting families at a very young age.

- iv. The Cummulative distribution function
    - This would be used to find out the probability that a random individual chosen from Uganda's population is below the age of 18. This information can be used the Ministry of Education & Sports to plan for an effective academic year.

- v. Poisson distribution
   - This would be used to estimate the number of births occuring per day based on the regions with a higher female to male sex region.

# PRACTICAL SECTION

In [ ]:
import pandas as pd
#importing the necessary library

In [26]:
students_df = pd.read_csv("Students.csv", delimiter=";")
students_df
#loading the dataframe 


Marital status  Application mode  Application order  Course  \
0                  1                17                  5     171   
1                  1                15                  1    9254   
2                  1                 1                  5    9070   
3                  1                17                  2    9773   
4                  2                39                  1    8014   
...              ...               ...                ...     ...   
4419               1                 1                  6    9773   
4420               1                 1                  2    9773   
4421               1                 1                  1    9500   
4422               1                 1                  1    9147   
4423               1                10                  1    9773   

      Daytime/evening attendance\t  Previous qualification  \
0                                1                       1   
1                                1                       1   
2                                1                       1   
3                                1                       1   
4                                0                       1   
...                            ...                     ...   
4419                             1                       1   
4420                             1                       1   
4421                             1                       1   
4422                             1                       1   
4423                             1                       1   

      Previous qualification (grade)  Nacionality  Mother's qualification  \
0                              122.0            1                      19   
1                              160.0            1                       1   
2                              122.0            1                      37   
3                              122.0            1                      38   
4                              100.0            1                      37   
...                              ...          ...                     ...   
4419                           125.0            1                       1   
4420                           120.0          105                       1   
4421                           154.0            1                      37   
4422                           180.0            1                      37   
4423                           152.0           22                      38   

      Father's qualification  ...  Curricular units 2nd sem (credited)  \
0                         12  ...                                    0   
1                          3  ...                                    0   
2                         37  ...                                    0   
3                         37  ...                                    0   
4                         38  ...                                    0   
...                      ...  ...                                  ...   
4419                       1  ...                                    0   
4420                       1  ...                                    0   
4421                      37  ...                                    0   
4422                      37  ...                                    0   
4423                      37  ...                                    0   

      Curricular units 2nd sem (enrolled)  \
0                                       0   
1                                       6   
2                                       6   
3                                       6   
4                                       6   
...                                   ...   
4419                                    6   
4420                                    6   
4421                                    8   
4422                                    5   
4423                                    6   

      Curricular units 2nd sem (evaluations)  \
0                            

In [24]:
students_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4424 entries, 0 to 4423
Data columns (total 37 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Marital status                                  4424 non-null   int64  
 1   Application mode                                4424 non-null   int64  
 2   Application order                               4424 non-null   int64  
 3   Course                                          4424 non-null   int64  
 4   Daytime/evening attendance	                     4424 non-null   int64  
 5   Previous qualification                          4424 non-null   int64  
 6   Previous qualification (grade)                  4424 non-null   float64
 7   Nacionality                                     4424 non-null   int64  
 8   Mother's qualification                          4424 non-null   int64  
 9   Father's qualification                   

# 2. Probability of a student being "Enrolled" under the variable "Target"

In [30]:
# the description of the Target variable to find out the values under it.
students_df['Target'].describe()
students_df['Target'].unique()

array(['Dropout', 'Graduate', 'Enrolled'], dtype=object)

In [31]:
# we get the total number of variables under Target. this will be our sample space.
target_total=students_df['Target'].count()
target_total

np.int64(4424)

In [32]:
# we then get the total count of Enrolled students.
enrolled=students_df[students_df['Target'] == "Enrolled"]['Target'].count()
enrolled

np.int64(794)

In [33]:
#probability of a student being "Enrolled" under the variable "Target"
Enrolled_prob= enrolled/target_total
Enrolled_prob


np.float64(0.1794755877034358)

- The probability that a student is enrolled is 17.94%

# 3. the probability of a student being a graduate with a marital status of 1

In [46]:
# first filter the data for a student that is a graduate whose marital status is 1
filtered_data=(students_df['Marital status'] == 1) & (students_df['Target']=='Graduate')


In [48]:
#the number of outcomes selecting a graduate student whose marital status is 1
graduate_1_count = students_df[filtered_data]["Target"].count()
graduate_1_count

np.int64(2015)

In [49]:
#the number of people with a marital status of 1
no_1 = students_df[students_df['Marital status'] == 1]['Marital status'].count()
no_1

np.int64(3919)

In [50]:
# calculate the probability of a student being a graduate and a marital status of 1
Graduate_prob=graduate_1_count/no_1
Graduate_prob

np.float64(0.514161775963256)

- the probability of a student being a graduate whose marital status is 1 is 51.42%

# 4. Marginal probability of variable "age at enrollment"

In [68]:
# the marginal probability is the probability of a single event happening irrespecive of the outcomes of the other related events.
# we ge the frequency of the unique values of this variable.
Age_count= students_df['Age at enrollment'].value_counts()
Age_count


Age at enrollment
18    1036
19     911
20     599
21     322
22     174
24     131
23     108
26      94
25      93
27      91
28      83
29      66
32      61
34      60
31      55
35      49
30      49
33      45
37      42
39      38
36      35
41      31
38      29
43      25
40      23
45      22
44      21
47      18
50      16
42      13
49      13
46      12
48      11
51       7
54       7
53       7
55       5
17       5
52       4
58       3
59       3
60       2
57       2
70       1
61       1
62       1
Name: count, dtype: int64

In [69]:
#We then calculate the marginal probability by dividing by the total count.
Marginal_prob=Age_count/len(students_df)
Marginal_prob

Age at enrollment
18    0.234177
19    0.205922
20    0.135398
21    0.072785
22    0.039331
24    0.029611
23    0.024412
26    0.021248
25    0.021022
27    0.020570
28    0.018761
29    0.014919
32    0.013788
34    0.013562
31    0.012432
35    0.011076
30    0.011076
33    0.010172
37    0.009494
39    0.008590
36    0.007911
41    0.007007
38    0.006555
43    0.005651
40    0.005199
45    0.004973
44    0.004747
47    0.004069
50    0.003617
42    0.002939
49    0.002939
46    0.002712
48    0.002486
51    0.001582
54    0.001582
53    0.001582
55    0.001130
17    0.001130
52    0.000904
58    0.000678
59    0.000678
60    0.000452
57    0.000452
70    0.000226
61    0.000226
62    0.000226
Name: count, dtype: float64

# 5. Probability that the "Age at enrollment" is 18

In [54]:
# we get the number of people that are 18 under the age of enrollment.
age18_count=students_df[students_df['Age at enrollment'] == 18]['Age at enrollment'].count()
age18_count


np.int64(1036)

In [55]:
# the total count of age at enrollment
age_enroll=students_df['Age at enrollment'].count()
age_enroll

np.int64(4424)

In [60]:
# the probability that age at enrollment is 18
age_18_prob= age18_count/age_enroll
age_18_prob


np.float64(0.23417721518987342)

- the probability that the age at enrollment is 18 is 23.42%